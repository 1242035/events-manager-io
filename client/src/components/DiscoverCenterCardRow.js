import React from 'react';
import { connect } from 'react-redux';
import DiscoverCenterCard from '../components/DiscoverCenterCard';

class DiscoverCenterCardRow extends React.Component {
  constructor(props) {
    super(props);

    let centers = [...this.props.center.centers];
    const centerCardElements = [];

    // If this component has a `type` prop with `near` value.
    if (this.props.type && this.props.type === 'near') {
      centers = centers.filter(eventCenter => eventCenter.location.trim().toLowerCase() === 'lagos');
    }

    centers.forEach((center) => {
      // Get details of each center.
      const {
        imageUrl, name, type, location,
      } = center;

      // Create centerCard elements and push them into centerCardElements
      centerCardElements.push(<div className="col-6 col-md-4 col-lg-3"><DiscoverCenterCard imageUrl={imageUrl} name={name} type={type} location={location} /></div>);
    });

    this.state = { centerCardElements };
  }

  render() {
    return (
      <div className="row io-top-row">{this.state.centerCardElements}</div>
    );
  }
}

const mapStateToProps = ({ center }) => ({ center });


export default connect(
  mapStateToProps,
  {},
)(DiscoverCenterCardRow);
