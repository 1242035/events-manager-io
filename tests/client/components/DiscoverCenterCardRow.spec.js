import React from 'react';
import { Provider } from 'react-redux';
import { shallow, mount } from 'enzyme';
import DiscoverCenterCardRow from '../../../client/src/components/DiscoverCenterCardRow';
import mockStore from '../mocks/storeMock';
import { mockAxios } from '../mocks/axiosMock';

describe('DiscoverCenterCardRow', () => {
  let wrapper;
  let store;
  mockAxios();

  beforeAll(() => {
    store = mockStore();
    localStorage.setItem('user.token', 'token');
    localStorage.setItem('user.admin', 'true');
  });

  afterAll(() => {
    localStorage.setItem('user.token', 'undefined');
    localStorage.setItem('user.admin', 'undefined');
  });

  it('Should maintain existing snapshot', () => {
    wrapper = shallow(<DiscoverCenterCardRow store={store} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('Should render connected component', () => {
    wrapper = mount(<Provider store={store}><DiscoverCenterCardRow /></Provider>);
    expect(wrapper.length).toEqual(1);
  });
});
