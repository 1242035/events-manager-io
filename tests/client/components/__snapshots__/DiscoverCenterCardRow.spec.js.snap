// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DiscoverCenterCardRow Should maintain existing snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
<<<<<<< HEAD
  Symbol(enzyme.__unrendered__): <Provider
=======
  Symbol(enzyme.__unrendered__): <Connect(DiscoverCenterCardRow)
>>>>>>> 3a61c4e8b27f096ee0a69bc247d734a3e37bdf4b
    store={
      Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      }
    }
<<<<<<< HEAD
  >
    <Connect(DiscoverCenterCardRow)
      showViewCenterModal={[Function]}
    />
  </Provider>,
=======
  />,
>>>>>>> 3a61c4e8b27f096ee0a69bc247d734a3e37bdf4b
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
<<<<<<< HEAD
      "showViewCenterModal": [Function],
=======
      "center": Object {
        "centers": Array [],
        "message": "",
      },
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
      "storeSubscription": Subscription {
        "listeners": Object {
          "clear": [Function],
          "get": [Function],
          "notify": [Function],
          "subscribe": [Function],
        },
        "onStateChange": [Function],
        "parentSub": undefined,
        "store": Object {
          "clearActions": [Function],
          "dispatch": [Function],
          "getActions": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
        },
        "unsubscribe": [Function],
      },
>>>>>>> 3a61c4e8b27f096ee0a69bc247d734a3e37bdf4b
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
<<<<<<< HEAD
        "showViewCenterModal": [Function],
=======
        "center": Object {
          "centers": Array [],
          "message": "",
        },
        "store": Object {
          "clearActions": [Function],
          "dispatch": [Function],
          "getActions": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
        },
        "storeSubscription": Subscription {
          "listeners": Object {
            "clear": [Function],
            "get": [Function],
            "notify": [Function],
            "subscribe": [Function],
          },
          "onStateChange": [Function],
          "parentSub": undefined,
          "store": Object {
            "clearActions": [Function],
            "dispatch": [Function],
            "getActions": [Function],
            "getState": [Function],
            "replaceReducer": [Function],
            "subscribe": [Function],
          },
          "unsubscribe": [Function],
        },
>>>>>>> 3a61c4e8b27f096ee0a69bc247d734a3e37bdf4b
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
